### ISSUE TITLE
Mod to make client think it has all recipes unlocked.

### ISSUE THREAD

**Issue by GlitchHead** [ImpactDevelopment/ImpactClient#460]
It would be awesome to have a mod that unlocks all recipes I understand it is done server-side but you can just have a mod make the client think it has them all unlocked. This would make crafting so easy with the search feature in the crafting table.

**Comment by ZeroMemes** [comment]
Not sure if this would be possible, the server probably tracks what you actually have unlocked. Will look into.

**Comment by GlitchHead** [comment]
@ZeroMemes if you know all of the default recipes on minecraft you can craft anything without having them in your "unlocked" recipes. So for servers that don't have custom crafting you could just have it appear on your client that you have everything unlocked and on the server you will just unlock default recipes once you use them.

**Comment by LeafHacker** [comment]
Good point, because all the recipe book does is put stuff in the crafting grid, and recipes are stored in game (are they resources yet in 1.12?) there's no reason we can't pretend to have all recipes.

**Comment by firestar9114** [comment]
I came here to request this, but found this post.  It is so annoying when I am trying to remember what I need to make an item, but because I haven't picked up some random combination of stuff I have to leave the game and go to the wiki.  Really, really annoying.  Especially when the book gets reset for some reason(reinstall, new world, new server, whatever).

**Comment by Thunder33345** [comment]
also most server dont have restricted crafting mode on
it's mostly visual so it's annoying you have to discover them by putting in manually
